pygotham/frontend/talks.py

```diff
@@ -100,6 +100,13 @@ def proposal(pk=None):
         db.session.add(talk)
         db.session.commit()
 
+        speaker = Speaker(
+            user_id=current_user.id,
+            talk_id=talk.id,
+        )
+        db.session.add(speaker)
+        db.session.commit()
```

Comment:
We should create the speaker and add it to the session before calling ``db.session.commit()`` above.

1. This will make sure that we're not making extra commits when not necessary.
2. Additionally, it will prevent talks from being submitted without users (here, `Speaker` instances) assigned to them; if something fails causing that second commit to error, we'll want to make sure that we don't create an "orphaned" talk in the database with no user to take credit for it.

Finally, when removing the commit that happens between speaker and talk creation, `talk_id=talk.id` should change to `talk=talk`. SQLAlchemy is smart enough to track commit dependency and assign the correct id on the fly, so we don't need to keep track of ids and can work with high level Python objects.